presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/file/PartitionMetadata.java

```diff
@@ -90,10 +88,7 @@ public PartitionMetadata(Table table, PartitionWithStatistics partitionWithStati
         this.columns = partition.getColumns();
         this.parameters = updateStatisticsParameters(partition.getParameters(), statistics.getBasicStatistics());
 
-        StorageFormat tableFormat = partition.getStorage().getStorageFormat();
-        storageFormat = Arrays.stream(HiveStorageFormat.values())
-                .filter(format -> tableFormat.equals(StorageFormat.fromHiveStorageFormat(format)))
-                .findFirst();
+        storageFormat = Optional.of(partition.getStorage().getStorageFormat());
```

Comment:
Looks like this is always present, why have the storageFormat as optional then ?

Reply:
Right. The optional here is unnecessary. Have pruned.